File: backend/StrategyGame.Model/Entities/Models/Egyseg.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace StrategyGame.Model.Entities.Models
{
    public abstract class Egyseg
    {
        [Key]
        public int EgysegId { get; set; }
        public int Tamadas { get; set; }
        public int Vedekezes { get; set; }
        public int Ar { get; set; }
        public int Zsold { get; set; }
        public int Ellatas { get; set; }
    }

    public class RohamFoka : Egyseg
    {

    }

    public class CsataCsiko : Egyseg
    {

    }

    public class LezerCapa : Egyseg
    {

    }
}

File: frontend/src/app/features/battle/models/egyseg.ts
export interface Egyseg {
  tamadas: number;
  vedekezes: number;
  ar: number;
  zsold: number;
  ellatas: number;
  mennyiseg: number;
  tipus: string;
}

export enum EgysegType {
  foka = 'RohamFoka',
  capa = 'LezerCapa',
  csiko = 'CsataCsiko'
}
